lib/functions/aeminitiateupload.js

```diff
@@ -104,11 +106,31 @@ class AEMInitiateUpload extends AsyncGeneratorFunction {
             const completeUrl = new URL(initiateResponse.completeURI, folderUrl);
             for (let i = 0; i < assets.length; ++i) {
                 const transferAsset = assets[i];
-                const { minPartSize, maxPartSize, uploadURIs, uploadToken } = files[i];
-                // uploadToken is opaque to us, we just need to send it back with the completeUpload
+                const { minPartSize, maxPartSize, uploadURIs, mimeType, uploadToken } = files[i];
+
+                // validate response from the server
+                // note: uploadToken is opaque to us, we just need to send it back with the completeUpload
                 if (!Number.isFinite(minPartSize) || !Number.isFinite(maxPartSize) || !Array.isArray(uploadURIs) || (uploadURIs.length === 0)) {
                     throw Error(`invalid multi-part information for ${transferAsset.target.url}: ${JSON.stringify(files[i])}`);
                 }
+                if (mimeType && typeof mimeType !== "string") {
+                    throw Error(`invalid mime-type for ${transferAsset.target.url}: ${JSON.stringify(mimeType)}`);
```

Comment:
will `JSON.stringify` throw if `mimetype` is not a JSON?

also, do you need more validation that the mimetype is a valid mimetype?

Reply:
It doesn't: 

```
> JSON.stringify(null)
'null'
> JSON.stringify(123)
'123'
> JSON.stringify("abc")
'"abc"'
> JSON.stringify({})
'{}'
```